test/metabase/api/user_test.clj

```diff
@@ -95,49 +97,79 @@
   [group-or-ids]
   (map (fn [group-or-id] {:id (u/the-id group-or-id)}) group-or-ids))
 
-(deftest user-recipients-list-test
-  (testing "GET /api/user/recipients"
+(deftest user-recipients-list-oss-test
+  (testing "GET /api/user/recipients without :email-restrict-recipients feature"
     (mt/with-non-admin-groups-no-root-collection-perms
       (let [crowberto "crowberto@metabase.com"
             lucky     "lucky@metabase.com"
             rasta     "rasta@metabase.com"]
-        (testing "Returns all users when user-visibility is all users"
-          (mt/with-temporary-setting-values [user-visibility :all]
-            (is (= [crowberto lucky rasta]
-                   (->> ((mt/user-http-request :rasta :get 200 "user/recipients") :data)
-                        (filter mt/test-user?)
-                        (map :email))))))
-        (testing "Returns all users when admin"
-          (mt/with-temporary-setting-values [user-visibility "none"]
-            (is (= [crowberto lucky rasta]
-                   (->> ((mt/user-http-request :crowberto :get 200 "user/recipients") :data)
-                        (filter mt/test-user?)
-                        (map :email))))))
-        (testing "Returns users in the group when user-visibility is same group"
-          (mt/with-temporary-setting-values [user-visibility :group]
-            (mt/with-temp* [PermissionsGroup           [{group-id :id} {:name "Test delete group"}]
-                            PermissionsGroupMembership [_ {:user_id (mt/user->id :rasta) :group_id group-id}]
-                            PermissionsGroupMembership [_ {:user_id (mt/user->id :crowberto) :group_id group-id}]]
-              (is (= [crowberto rasta]
+
+        (testing "return all users for anyone"
+          (is (= [crowberto lucky rasta]
+                 (->> (:data (mt/user-http-request :rasta :get 200 "user/recipients"))
+                      (filter mt/test-user?)
+                      (map :email))))
+
+          (is (= [crowberto lucky rasta]
+                 (->> (:data (mt/user-http-request :rasta :get 200 "user/recipients"))
+                      (filter mt/test-user?)
+                      (map :email)))))
+
+        (testing "not affected by the visibility setting"
+          (doseq [visibility-value [:all :group :none]]
+            (is (= :all (api.user/user-visibility)))
```

Comment:
I think you mean to put this `is` inside the `with-temporary-setting-values` body.